src/Microsoft.AspNet.OData/Adapters/WebApiRequestMessage.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Web.Http;
+using System.Web.Http.Routing;
+using Microsoft.AspNet.OData.Batch;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Formatter.Deserialization;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Adapters
+{
+    /// <summary>
+    /// Adapter class to convert Asp.Net WebApi request message to OData WebApi.
+    /// </summary>
+    internal class WebApiRequestMessage : IWebApiRequestMessage
+    {
+        /// <summary>
+        /// The inner request wrapped by this instance.
+        /// </summary>
+        internal HttpRequestMessage innerRequest;
+
+        /// <summary>
+        /// Initializes a new instance of the WebApiRequestMessage class.
+        /// </summary>
+        /// <param name="request">The inner request.</param>
+        public WebApiRequestMessage(HttpRequestMessage request)
+        {
+            if (request == null)
+            {
+                throw Error.ArgumentNull("request");
+            }
+
+            this.innerRequest = request;
+
+            HttpRequestMessageProperties context = request.ODataProperties();
+            if (context != null)
+            {
+                this.Context = new WebApiContext(context);
+            }
+
+            UrlHelper uriHelper = request.GetUrlHelper();
+            if (uriHelper != null)
+            {
+                this.UrlHelper = new WebApiUrlHelper(uriHelper);
+            }
+
+            HttpConfiguration configuration = request.GetConfiguration();
+            if (configuration != null)
+            {
+                this.Options = new WebApiOptions(configuration);
+            }
+        }
+
+        /// <summary>
+        /// Gets the contents of the HTTP message. 
+        /// </summary>
+        public IWebApiContext Context { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating if this is a count request.
+        /// </summary>
+        /// <returns></returns>
+        public bool IsCountRequest()
+        {
+            return ODataCountMediaTypeMapping.IsCountRequest(this.innerRequest);
+        }
+
+        /// <summary>
+        /// Gets the HTTP method used by the HTTP request message.
+        /// </summary>
+        public ODataRequestMethod Method
+        {
+            get
+            {
+                string method = this.innerRequest.Method.ToString().ToUpperInvariant();
```

Comment:
nit: is `.ToUpperInvariant()` necessary if you're ignoring case later below?

Reply:
Not needed, it's a left-over from a time when a string was returned from this property instead of an enum.